File: packetCapture/src/main/java/com/ninpou/packetcapture/core/vpn/VpnServiceProxy.java
package com.ninpou.packetcapture.core.vpn;

import java.lang.ref.WeakReference;
import java.net.DatagramSocket;
import java.net.Socket;

public class VpnServiceProxy {
    private static WeakReference<VpnServiceImpl> vpnService = null;

    public static void setVpnService(VpnServiceImpl vpnService) {
        VpnServiceProxy.vpnService = new WeakReference<>(vpnService);
    }

    public static boolean protect(Socket socket) {
        if (vpnService == null || vpnService.get() == null) return false;
        return vpnService.get().protect(socket);
    }

    public static boolean protect(DatagramSocket socket) {
        if (vpnService == null || vpnService.get() == null) return false;
        return vpnService.get().protect(socket);
    }

    public static int getMtu() {
        return VpnServiceImpl.MTU;
    }

    public static String getAddress() {
        return VpnServiceImpl.ADDRESS;
    }

    public static long getStartTime() {
        if (vpnService == null || vpnService.get() == null) return -1;
        return vpnService.get().getStartTime();
    }
}

File: packetCapture/src/main/java/com/ninpou/packetcapture/core/vpn/VpnServiceProxy.kt
package com.ninpou.packetcapture.core.vpn

import java.lang.ref.WeakReference
import java.net.DatagramSocket
import java.net.Socket

object VpnServiceProxy {
    private var vpnService: WeakReference<VpnServiceImpl?>? = null
    @JvmStatic
    fun setVpnService(vpnService: VpnServiceImpl?) {
        VpnServiceProxy.vpnService = WeakReference(vpnService)
    }

    @JvmStatic
    fun protect(socket: Socket?): Boolean {
        return if (vpnService == null || vpnService!!.get() == null) false else vpnService!!.get()!!.protect(socket)
    }

    @JvmStatic
    fun protect(socket: DatagramSocket?): Boolean {
        return if (vpnService == null || vpnService!!.get() == null) false else vpnService!!.get()!!.protect(socket)
    }

    @JvmStatic
    val mtu: Int
        get() = VpnServiceImpl.MTU
    @JvmStatic
    val address: String
        get() = VpnServiceImpl.ADDRESS
    @JvmStatic
    val startTime: Long
        get() = if (vpnService == null || vpnService!!.get() == null) -1 else vpnService!!.get()!!.startTime
}